.github/workflows/CI.yml

```diff
@@ -281,27 +286,26 @@ jobs:
       if: ${{ github.event_name == 'push' }}
 
     - name: Download Artifacts
-      uses: actions/download-artifact@v2
+      uses: actions/download-artifact@v4
       with:
-        path: ${{ github.workspace }} # This will download all files
+        path: fpm-cd-artifacts
```

Comment:
Here we should be more selective of what we are downloading from the workflow.

Reply:
These are the artifacts we download, which ones do you think we should skip?

- fpm-installer-gcc-11 (ID: 1582985908, Size: 28334370)
- fpm-installer-gcc-10 (ID: 1582985888, Size: 28334503)
- fpm-installer-gcc-12 (ID: 1582985862, Size: 28334981)
- fpm-a663b5d-windows-x86_64-gcc-10.exe (ID: 1582985248, Size: [19](https://github.com/fortran-lang/fpm/actions/runs/9436137530/job/25990571160?pr=985#step:4:20)61036)
- fpm-a663b5d-windows-x86_64-gcc-12.exe (ID: 1582985148, Size: 1961152)
- fpm-a663b5d-windows-x86_64-gcc-11.exe (ID: 1582985108, Size: 1961106)
- fpm-a663b5d-macos-x86_64-gcc-12 (ID: 158298[20](https://github.com/fortran-lang/fpm/actions/runs/9436137530/job/25990571160?pr=985#step:4:21)39, Size: 717370)
- fpm-a663b5d-macos-x86_64-gcc-11 (ID: 1582981934, Size: 711913)
- fpm-a663b5d-macos-x86_64-gcc-10 (ID: 1582981631, Size: 6946[22](https://github.com/fortran-lang/fpm/actions/runs/9436137530/job/25990571160?pr=985#step:4:23))
- fpm-a663b5d-linux-x86_64-gcc-12 (ID: 1582979783, Size: 2304462)
- fpm-a663b5d-linux-x86_64-gcc-11 (ID: 1582979563, Size: 2283349)
- fpm-a663b5d-linux-x86_64-gcc-10 (ID: 1582979285, Size: 2292735)

Reply:
I see, they might have the commit hash in the name, which makes it a bit tricky to define the names without extra logic. We can use the latest GCC version for the distribution.

Reply:
ok, latest GCC version sounds good to me.
FYI I was thinking in a subsequent PR to move the code of upload-artifacts from CI.yml and release.yml into it's own reusable workflow to remove the duplication.

Reply:
I forgot to normalise the `-gcc-12` part of the name. I'll try and fix that sometime today.